File: app/src/main/java/com/yanxing/ui/CommonAdapterFragment.java
package com.yanxing.ui;

import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.StaggeredGridLayoutManager;
import android.view.View;
import android.view.ViewGroup;
import android.widget.GridView;
import android.widget.ListView;
import android.widget.TextView;

import com.yanxing.adapterlibrary.CommonAdapter;
import com.yanxing.adapterlibrary.RecyclerViewAdapter;
import com.yanxing.adapterlibrary.ViewHolder;
import com.yanxing.base.BaseFragment;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;

/**
 * adapterlibrary测试
 * Created by lishuangxiang on 2016/1/20.
 */
public class CommonAdapterFragment extends BaseFragment {

    @BindView(R.id.listView)
    ListView mListView;

    @BindView(R.id.gridView)
    GridView mGridView;

    @BindView(R.id.recyclerView)
    RecyclerView mRecyclerView;

    @Override
    protected int getLayoutResID() {
        return R.layout.fragment_common_adapter;
    }

    @Override
    protected void afterInstanceView() {
        final List<Integer> list = new ArrayList<>();
        for (int i = 0; i < 7; i++) {
            list.add(i);

        }
        CommonAdapter<Integer> adapter = new CommonAdapter<Integer>(list, R.layout.list_dialog_textview) {
            @Override
            public void onBindViewHolder(ViewHolder viewHolder, int position) {
                viewHolder.setText(R.id.text, String.valueOf(mDataList.get(position)));
            }
        };
        //ListView
        mListView.setAdapter(adapter);
        //GridView
        mGridView.setAdapter(adapter);
        //RecycleView
        mRecyclerView.setLayoutManager(new StaggeredGridLayoutManager(4, StaggeredGridLayoutManager.VERTICAL));
        RecyclerViewAdapter recyclerViewAdapter = new RecyclerViewAdapter<Integer>(list, R.layout.adapter_recycler_view) {

            @Override
            public void onBindViewHolder(RecyclerViewAdapter.MyViewHolder holder, final int position) {
                TextView textView = (TextView) holder.findViewById(R.id.text);
                //瀑布流，动态设置item大小
                ViewGroup.LayoutParams lp = textView.getLayoutParams();
                lp.height = (int) (50 + Math.random() * 300);
                textView.setLayoutParams(lp);
                textView.setText(String.valueOf(list.get(position)));
                holder.itemView.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        showToast(getString(R.string.you_click) + (position + 1) + getString(R.string.ge));
                    }
                });
            }
        };
        mRecyclerView.setAdapter(recyclerViewAdapter);
    }
}

File: app/src/main/java/com/yanxing/ui/CommonAdapterFragment.kt
package com.yanxing.ui

import androidx.recyclerview.widget.StaggeredGridLayoutManager
import android.view.View
import android.widget.TextView

import com.yanxing.base.BaseFragment
import com.yanxing.adapter.RecyclerViewAdapter

import java.util.ArrayList

import kotlinx.android.synthetic.main.fragment_common_adapter.*

/**
 * adapterlibrary测试
 * Created by lishuangxiang on 2016/1/20.
 */
class CommonAdapterFragment : BaseFragment() {

    override fun getLayoutResID(): Int {
        return R.layout.fragment_common_adapter
    }

    override fun afterInstanceView() {
        val list = ArrayList<Int>()
        for (i in 0..6) {
            list.add(i)

        }
        //RecycleView
        recyclerView.layoutManager = StaggeredGridLayoutManager(4, StaggeredGridLayoutManager.VERTICAL)
        val recyclerViewAdapter = object : RecyclerViewAdapter<Int>(list, R.layout.adapter_recycler_view) {

            override fun onBindViewHolder(holder: RecyclerViewAdapter.MyViewHolder, position: Int) {
                super.onBindViewHolder(holder, position)
                val textView = holder.findViewById<View>(R.id.text) as TextView
                //瀑布流，动态设置item大小
                val lp = textView.layoutParams
                lp.height = (50 + Math.random() * 300).toInt()
                textView.layoutParams = lp
                textView.text = list[position].toString()
            }
        }
        recyclerViewAdapter.setOnItemClick { viewHolder, position ->
            showToast(getString(R.string.you_click) + (position + 1) + getString(
                    R.string.ge))
        }
        recyclerView.adapter = recyclerViewAdapter
    }
}
